lib/events/athena/integration_test.go

```diff
@@ -0,0 +1,391 @@
+		glueClient := glue.NewFromConfig(ac.awsCfg)
+		_, err := glueClient.GetDatabase(ctx, &glue.GetDatabaseInput{
+			Name: aws.String(ac.database),
+		})
+		if err != nil {
+			var notFound *glueTypes.EntityNotFoundException
+			if errors.As(err, &notFound) {
+				_, createErr := glueClient.CreateDatabase(ctx, &glue.CreateDatabaseInput{
+					DatabaseInput: &glueTypes.DatabaseInput{
+						Name: aws.String(ac.database),
+					},
+				})
+				require.NoError(t, createErr)
+			} else {
+				require.NoError(t, err)
+			}
+		}
+	})
+
+	t.Run("create athena table", func(t *testing.T) {
+		athenaClient := athena.NewFromConfig(ac.awsCfg)
+
+		startQueryExecResp, err := athenaClient.StartQueryExecution(ctx, &athena.StartQueryExecutionInput{
+			QueryString: aws.String(fmt.Sprintf(createTableQuery, ac.tablename, ac.s3eventsLocation, ac.s3eventsLocation)),
+			ResultConfiguration: &athenaTypes.ResultConfiguration{
+				OutputLocation: aws.String(ac.s3resultsLocation),
+			},
+			QueryExecutionContext: &athenaTypes.QueryExecutionContext{
+				Database: aws.String(ac.database),
+			},
+		})
+		require.NoError(t, err)
+
+		// querier is just used here to get helper fn waitForSuccess.
+		q := querier{
+			athenaClient: athenaClient,
+			querierConfig: querierConfig{
+				getQueryResultsInterval: 100 * time.Millisecond,
+				clock:                   ac.clock,
+			},
+		}
+		err = q.waitForSuccess(ctx, aws.ToString(startQueryExecResp.QueryExecutionId))
+		require.NoError(t, err)
+	})
+}
+
+// createTableQuery is query used to create athena table using parquet on s3.
+// Right now only hardcoded in integration tests, in future it may be moved
+// to athena main file if we decide to create table on demand from teleport.
+var createTableQuery = `
+CREATE EXTERNAL TABLE %s (
+	uid string,
+	session_id string,
+	event_type string,
+	event_time timestamp,
+	event_data string
+  )
+  PARTITIONED BY (
+   event_date DATE
+  )
+  ROW FORMAT SERDE 'org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe'
+  STORED AS INPUTFORMAT 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat'
+  OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat'
+  LOCATION "%s/"
+  TBLPROPERTIES (
+   "projection.enabled" = "true",
+   "projection.event_date.type" = "date",
+   "projection.event_date.format" = "yyyy-MM-dd",
+   "projection.event_date.range" = "NOW-4YEARS,NOW",
+   "projection.event_date.interval" = "1",
+   "projection.event_date.interval.unit" = "DAYS",
+   "storage.location.template" = "%s/${event_date}/",
+   "classification" = "parquet",
+   "parquet.compression" = "SNAPPY"
+  )
+`
+
+func (ac *athenaContext) Close(t *testing.T, ctx context.Context) {
+	t.Log("Cleaning up infra")
+	ac.log.Close()
+
+	sqsClient := sqs.NewFromConfig(ac.awsCfg)
+
+	_, err := sqsClient.DeleteQueue(ctx, &sqs.DeleteQueueInput{
+		QueueUrl: aws.String(ac.infraOutputs.queueURL),
+	})
+	require.NoError(t, err)
```

Comment:
Should we proceed proceed with cleanup even this call failed  ?

Reply:
good point, I will rework it.

Reply:
reworked in https://github.com/gravitational/teleport/pull/25248/commits/584f45e117d38dd5936c89069e8a1850b1045ed0